// @azure/arm-containerinstance/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ContainerNetworkProtocol} that the service accepts. */
export var KnownContainerNetworkProtocol;
(function (KnownContainerNetworkProtocol) {
    KnownContainerNetworkProtocol["TCP"] = "TCP";
    KnownContainerNetworkProtocol["UDP"] = "UDP";
})(KnownContainerNetworkProtocol || (KnownContainerNetworkProtocol = {}));
/** Known values of {@link GpuSku} that the service accepts. */
export var KnownGpuSku;
(function (KnownGpuSku) {
    KnownGpuSku["K80"] = "K80";
    KnownGpuSku["P100"] = "P100";
    KnownGpuSku["V100"] = "V100";
})(KnownGpuSku || (KnownGpuSku = {}));
/** Known values of {@link Scheme} that the service accepts. */
export var KnownScheme;
(function (KnownScheme) {
    KnownScheme["Http"] = "http";
    KnownScheme["Https"] = "https";
})(KnownScheme || (KnownScheme = {}));
/** Known values of {@link ContainerGroupRestartPolicy} that the service accepts. */
export var KnownContainerGroupRestartPolicy;
(function (KnownContainerGroupRestartPolicy) {
    KnownContainerGroupRestartPolicy["Always"] = "Always";
    KnownContainerGroupRestartPolicy["OnFailure"] = "OnFailure";
    KnownContainerGroupRestartPolicy["Never"] = "Never";
})(KnownContainerGroupRestartPolicy || (KnownContainerGroupRestartPolicy = {}));
/** Known values of {@link ContainerGroupNetworkProtocol} that the service accepts. */
export var KnownContainerGroupNetworkProtocol;
(function (KnownContainerGroupNetworkProtocol) {
    KnownContainerGroupNetworkProtocol["TCP"] = "TCP";
    KnownContainerGroupNetworkProtocol["UDP"] = "UDP";
})(KnownContainerGroupNetworkProtocol || (KnownContainerGroupNetworkProtocol = {}));
/** Known values of {@link ContainerGroupIpAddressType} that the service accepts. */
export var KnownContainerGroupIpAddressType;
(function (KnownContainerGroupIpAddressType) {
    KnownContainerGroupIpAddressType["Public"] = "Public";
    KnownContainerGroupIpAddressType["Private"] = "Private";
})(KnownContainerGroupIpAddressType || (KnownContainerGroupIpAddressType = {}));
/** Known values of {@link AutoGeneratedDomainNameLabelScope} that the service accepts. */
export var KnownAutoGeneratedDomainNameLabelScope;
(function (KnownAutoGeneratedDomainNameLabelScope) {
    KnownAutoGeneratedDomainNameLabelScope["Unsecure"] = "Unsecure";
    KnownAutoGeneratedDomainNameLabelScope["TenantReuse"] = "TenantReuse";
    KnownAutoGeneratedDomainNameLabelScope["SubscriptionReuse"] = "SubscriptionReuse";
    KnownAutoGeneratedDomainNameLabelScope["ResourceGroupReuse"] = "ResourceGroupReuse";
    KnownAutoGeneratedDomainNameLabelScope["Noreuse"] = "Noreuse";
})(KnownAutoGeneratedDomainNameLabelScope || (KnownAutoGeneratedDomainNameLabelScope = {}));
/** Known values of {@link OperatingSystemTypes} that the service accepts. */
export var KnownOperatingSystemTypes;
(function (KnownOperatingSystemTypes) {
    KnownOperatingSystemTypes["Windows"] = "Windows";
    KnownOperatingSystemTypes["Linux"] = "Linux";
})(KnownOperatingSystemTypes || (KnownOperatingSystemTypes = {}));
/** Known values of {@link LogAnalyticsLogType} that the service accepts. */
export var KnownLogAnalyticsLogType;
(function (KnownLogAnalyticsLogType) {
    KnownLogAnalyticsLogType["ContainerInsights"] = "ContainerInsights";
    KnownLogAnalyticsLogType["ContainerInstanceLogs"] = "ContainerInstanceLogs";
})(KnownLogAnalyticsLogType || (KnownLogAnalyticsLogType = {}));
/** Known values of {@link ContainerGroupSku} that the service accepts. */
export var KnownContainerGroupSku;
(function (KnownContainerGroupSku) {
    KnownContainerGroupSku["Standard"] = "Standard";
    KnownContainerGroupSku["Dedicated"] = "Dedicated";
})(KnownContainerGroupSku || (KnownContainerGroupSku = {}));
/** Known values of {@link ContainerInstanceOperationsOrigin} that the service accepts. */
export var KnownContainerInstanceOperationsOrigin;
(function (KnownContainerInstanceOperationsOrigin) {
    KnownContainerInstanceOperationsOrigin["User"] = "User";
    KnownContainerInstanceOperationsOrigin["System"] = "System";
})(KnownContainerInstanceOperationsOrigin || (KnownContainerInstanceOperationsOrigin = {}));
